Setup frontend test script (CI)

stevenwu2019 opened the pull request:
* Added ESLint and Jest to Expo
* Created automated testing CI script for frontend
   * Building is not included in script since we only have 30 free builds

emmaline01 reviewed (commented):
Github Actions looks pretty good - it clearly runs and passes haha

Some things I noticed:
 - in the Dronuts project we also used `npm-check` to check our dependencies are up-to-date. Should we do that for this project also?
 - After I deleted my local node_modules and reinstalled with `npm i` I had to locally install jest for the tests to run... I also had to install eslint, prettier, and typescript modules for the lint to run, and then the lint gave 219 warnings, all from prettier. Honestly I'm not sure what's happening. HOWEVER it did all work after I deleted my node_modules, reinstalled with `npm ci`, and tried again. I'm not too clear on the differences between the two install commands.
 - Jest isn't set up to run any backend tests right now. Should it be?

Also, would you mind updating the README for this project since we'll be done with the repo setup by the time of this PR's approval?

hachiyuki8 commented:
> Github Actions looks pretty good - it clearly runs and passes haha
> 
> Some things I noticed:
> 
> * in the Dronuts project we also used `npm-check` to check our dependencies are up-to-date. Should we do that for this project also?
> * After I deleted my local node_modules and reinstalled with `npm i` I had to locally install jest for the tests to run... I also had to install eslint, prettier, and typescript modules for the lint to run, and then the lint gave 219 warnings, all from prettier. Honestly I'm not sure what's happening. HOWEVER it did all work after I deleted my node_modules, reinstalled with `npm ci`, and tried again. I'm not too clear on the differences between the two install commands.
> * Jest isn't set up to run any backend tests right now. Should it be?
> 
> Also, would you mind updating the README for this project since we'll be done with the repo setup by the time of this PR's approval?

re: backend testing part: backend will have a separate testing module and hence eventually (once I set it up) a separate script to run it

stevenwu2019 commented:
> * in the Dronuts project we also used `npm-check` to check our dependencies are up-to-date. Should we do that for this project also?

> Also, would you mind updating the README for this project since we'll be done with the repo setup by the time of this PR's approval?

Implemented the above feedback

> * After I deleted my local node_modules and reinstalled with `npm i` I had to locally install jest for the tests to run... I also had to install eslint, prettier, and typescript modules for the lint to run, and then the lint gave 219 warnings, all from prettier. Honestly I'm not sure what's happening. HOWEVER it did all work after I deleted my node_modules, reinstalled with `npm ci`, and tried again. I'm not too clear on the differences between the two install commands.

According to this [thread](https://stackoverflow.com/questions/52499617/what-is-the-difference-between-npm-install-and-npm-ci), `npm i` is basically for use during development and `npm ci` is for use during automated testing. `npm i` should install more than `npm ci`, so I'm not sure how it was still missing ESLint and other modules when `npm ci` was not...
`npm ci` also deletes `node_modules` and reinstalls each time fyi, so it takes care of that for you

hachiyuki8 reviewed (approved):
lgtm
